Sources/TuistLoader/Models+ManifestMappers/CoreDataModel+ManifestMapper.swift

```diff
@@ -12,7 +12,63 @@ extension TuistCore.CoreDataModel {
     static func from(manifest: ProjectDescription.CoreDataModel, generatorPaths: GeneratorPaths) throws -> TuistCore.CoreDataModel {
         let modelPath = try generatorPaths.resolve(path: manifest.path)
         let versions = FileHandler.shared.glob(modelPath, glob: "*.xcdatamodel")
-        let currentVersion = manifest.currentVersion
+
+        var currentVersion: String!
+        if let hardcodedVersion = manifest.currentVersion {
+            currentVersion = hardcodedVersion
+        } else {
+            currentVersion = try CoreDataVersionExtractor.version(fromVersionFileAtPath: modelPath)
+        }
         return CoreDataModel(path: modelPath, versions: versions, currentVersion: currentVersion)
     }
 }
+
+/// Extract version from .xccurrentversion file.
+private final class CoreDataVersionExtractor {
+
+    /// Extract version from .xccurrentversion file
+    /// - Parameter filePath: absolute path to Model.xcdatamodel
+    /// - Throws: In case can not find the .xcurrentversion file.
+    /// - Returns: Current version of data model.
+    static func version(fromVersionFileAtPath filePath: AbsolutePath) throws -> String {
+        let url = URL(fileURLWithPath: filePath.pathString + "/.xccurrentversion")
```

Comment:
There's no need to turn `AbsolutePath` into another type that represents the same thing, `URL`.